pagerduty/resource_pagerduty_event_orchestration_path_global.go

```diff
@@ -0,0 +1,423 @@
+package pagerduty
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+var eventOrchestrationPathGlobalCatchAllActionsSchema = map[string]*schema.Schema{
+	"drop_event": {
+		Type:     schema.TypeBool,
+		Optional: true,
+	},
+	"suppress": {
+		Type:     schema.TypeBool,
+		Optional: true,
+	},
+	"suspend": {
+		Type:     schema.TypeInt,
+		Optional: true,
+	},
+	"priority": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"annotate": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"automation_action": {
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: eventOrchestrationAutomationActionSchema,
+		},
+	},
+	"severity": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validateEventOrchestrationPathSeverity(),
+	},
+	"event_action": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validateEventOrchestrationPathEventAction(),
+	},
+	"variable": {
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: eventOrchestrationPathVariablesSchema,
+		},
+	},
+	"extraction": {
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: eventOrchestrationPathExtractionsSchema,
+		},
+	},
+}
+
+var eventOrchestrationPathGlobalRuleActionsSchema = buildEventOrchestrationPathGlobalRuleActionsSchema()
+
+func buildEventOrchestrationPathGlobalRuleActionsSchema() map[string]*schema.Schema {
+	a := eventOrchestrationPathGlobalCatchAllActionsSchema
+	a["route_to"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Optional: true,
+	}
+
+	return a
+}
+
+func resourcePagerDutyEventOrchestrationPathGlobal() *schema.Resource {
```

Comment:
Although the schema for Global Orchestrations mostly overlaps with the Service Orchestration resource it's still slightly different (has `drop_event` but has no `pagerduty_automation_actions`). I thought that having both schemas inheriting from the same base schema would not make the code any more legible so i kept them separate. What I am sharing here is the Automation Action (webhook) schema that's more complex.